### PR TITLE
[Room Settings] Let the admin convert a forum in a private room 

### PR DESCRIPTION
**Pull request by yostyle**
#209

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt [hunk 1]
```diff
@@ -248,6 +293,34 @@ class RoomSettingsViewModel @AssistedInject constructor(@Assisted initialState:
...
+                // Turn on the encryption in this room (if this is not already done).
+                if (!room.isEncrypted()) room.enableEncryption()
+                // Remove the room from the room directory.
+                session.setRoomDirectoryVisibility(room.roomId, RoomDirectoryVisibility.PRIVATE)
+            } catch (failure: Throwable) {
+                postLoading(false)
+                _viewEvents.post(RoomSettingsViewEvents.Failure(failure))
+            }
+            // Update history visibility.
+            tryOrNull { room.updateHistoryReadability(RoomHistoryVisibility.INVITED) }
```
Comment (giomfo) [inline vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt:319]: During my tests, I observed an issue at the level of the definition of `enum class RoomHistoryVisibility `.
TchapV2 doesn't have the same definition as Element-Android

I realized this issue when I read (in the web client) this following message:
image

Comment (giomfo) [reply vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt:319]: the expected message is:
image

Comment (giomfo) [reply vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt:319]: here is the diff:
image

File: vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt [hunk 2]
```diff
@@ -150,7 +190,11 @@ class RoomSettingsViewModel @AssistedInject constructor(@Assisted initialState:
                                     && powerLevelsHelper.isUserAllowedToSend(session.myUserId, true,
                                     EventType.STATE_ROOM_GUEST_ACCESS),
                             canAddChildren = powerLevelsHelper.isUserAllowedToSend(session.myUserId, true,
-                                    EventType.STATE_SPACE_CHILD)
+                                    EventType.STATE_SPACE_CHILD),
+                            canRemoveFromRoomsDirectory = powerLevelsHelper.isUserAllowedToSend(session.myUserId, true, EventType.STATE_ROOM_THIRD_PARTY_INVITE)
```
[the remark this answers is not in the document]
Comment (yostyle) [reply vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt]: It's an error. I fixed that

File: vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt (hunk 1 above)
Comment (yostyle) [reply vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt:319]: It's an error during the rebase on element 1.1.9. I fixed it.

**Review by giomfo — APPROVED** [review]
LGTM, Tests ok

**Pull request merged**